cmd/puppeth/module_node.go

```diff
@@ -32,8 +32,24 @@ import (
 
 // nodeDockerfile is the Dockerfile required to run an Ethereum node.
 var nodeDockerfile = `
-FROM ethereum/client-go:latest
+{{if .Git}}
+	FROM golang:1.16-alpine as builder
 
+	RUN apk add --no-cache make gcc musl-dev linux-headers git
+
+	RUN git clone {{.GitRepo}} --branch {{.GitBranch}} --single-branch /go-ethereum
```

Comment:
It probably also makes sense to add `--depth=1` to not download the entire commit history, just the tip.

Reply:
Good idea.